src/Middleware/Diagnostics/src/ExceptionHandler/ExceptionHandlerMiddlewareImpl.cs

```diff
@@ -188,7 +188,12 @@ private async Task HandleException(HttpContext context, ExceptionDispatchInfo ed
                 }
                 else
                 {
-                    handled = await _problemDetailsService!.TryWriteAsync(new()
+                    if (_problemDetailsService is null)
```

Comment:
Oh, based on the thread with @Tratcher, maybe this just needs a better exception message?  I'm also curious about how the exception actually manifests.

Reply:
This gets caught below and logged, then the original exception is re-thrown. The server catches it and produces a blank 500.

Reply:
I placed it here because we only hit this situation if the exception handlers were not able to successfully process the exception. You don't necessarily run into this issue unless that's the case so if we were to move it to the constructor we would want a lower-severity notification to the user.